provider/kubernetes/annotations_parser.go

```diff
@@ -41,21 +41,25 @@ func getSliceAnnotation(meta *v1beta1.Ingress, name string) []string {
 	return value
 }
 
-func getMapAnnotation(meta *v1beta1.Ingress, name string) map[string]string {
-	value := make(map[string]string)
-	if annotation := meta.Annotations[name]; annotation != "" {
-		for _, v := range strings.Split(annotation, ",") {
-			pair := strings.Split(v, ":")
+func getMapAnnotation(meta *v1beta1.Ingress, labelName string) map[string]string {
+	if values, ok := meta.Annotations[labelName]; ok {
+		mapValue := make(map[string]string)
+
+		for _, parts := range strings.Split(values, ",") {
+			pair := strings.Split(parts, ":")
 			if len(pair) != 2 {
-				log.Debugf("Could not load annotation (%v) with value: %v, skipping...", name, pair)
+				log.Warnf("Could not load %q: %v, skipping...", labelName, pair)
 			} else {
-				value[pair[0]] = pair[1]
+				mapValue[pair[0]] = pair[1]
 			}
 		}
+
+		if len(mapValue) == 0 {
+			log.Errorf("Could not load %q, skipping...", labelName)
```

Comment:
I think the log messages could be clearer with regards to what the user did wrong, both in terms of wording and how we log.

My suggestion would be to do the following:

- Do a check against the length of `values` in line 47 and bail out immediately when it's zero along with a `log.Errorf("Missing value for annotation %q", values)`.
- Instead of logging an error right away in line 51, collect all illegal pairs first and print them in a comma-joint list inside a single `log.Warnf()`.

WDYT?

Reply:
I agree with the first point.
For the second point,it does not seem necessary because this create a complexity just for logging.

Reply:
Re: the second point: My reasoning is that we collect all illegal pairs and show them to the user in a more concise manner.

It'd be nice to have IMHO, though not strictly necessary.